File: advertising_sdk/src/main/java/com/adadapted/android/sdk/core/keywordintercept/Suggestion.java
package com.adadapted.android.sdk.core.keywordintercept;

public class Suggestion {
    private final String replacement;
    private final String icon;
    private final String tagLine;

    public Suggestion(final String replacement,
                      final String icon,
                      final String tagLine) {
        this.replacement = replacement;
        this.icon = icon;
        this.tagLine = tagLine;
    }

    public String getReplacement() {
        return replacement;
    }

    public String getIcon() {
        return icon;
    }

    public String getTagLine() {
        return tagLine;
    }
}

File: advertising_sdk/src/main/java/com/adadapted/android/sdk/core/keyword/Suggestion.kt
package com.adadapted.android.sdk.core.keyword

import com.adadapted.android.sdk.core.keyword.SuggestionTracker.suggestionPresented
import com.adadapted.android.sdk.core.keyword.SuggestionTracker.suggestionSelected

data class Suggestion(val searchId: String, private val term: InterceptTerm) {
    val termId: String = term.termId
    val name: String = term.replacement
    var presented: Boolean
    var selected: Boolean

    init {
        presented = false
        selected = false
    }

    fun presented() {
        if (!presented) {
            presented = true
            suggestionPresented(searchId, termId, name)
        }
    }

    fun selected() {
        if (!selected) {
            selected = true
            suggestionSelected(searchId, termId, name)
        }
    }
}